agent-control/src/opamp/remote_config/validators/signature/validator.rs

```diff
@@ -111,7 +111,7 @@ impl RemoteConfigValidator for SignatureValidator {
         };
 
         // Iterate over all remote configs and verify signatures
-        for (config_name, config_content) in opamp_remote_config.configs().0.iter() {
+        for (config_name, config_content) in opamp_remote_config.configs_iter() {
```

Comment:
I think the implemented behavior is alright although it can be a bit counter-intuitive: all configurations except the ones having `configAgent` as a prefix are ignored but their signature is checked anyway. In other words, if a configuration is included in the config-map, it needs to be properly signed regardless of its usage.